### ISSUE TITLE
Question ! Using with Nuxt

### ISSUE THREAD

**Issue by muscle-rhnkyr** [williamcruzme/vue-gates#4]
@williamcruzme thanks for super cool lib. I wonder where should I put 

```
const { data: permissions } = await axios.get('/api/permissions');
const { data: roles } = await axios.get('/api/roles');

this.$laravel.setPermissions(permissions);
this.$laravel.setRoles(roles);
```

while working with Nuxt?

Thanks in advance.

**Comment by williamcruzme** [comment]
@muscle-rhnkyr Thank you very much for your comment!

These days I will work on the support for Nuxt.js. I'll keep you informed right here.

Greetings.

**Comment by williamcruzme** [comment]
@muscle-rhnkyr I have published a new version **1.3.0** that includes support with Nuxt.js perfectly.

In the README.md you can find the instructions for the installation 🍻

This code should be put once you log in to get your permissions and roles. So when they are redirected you would only show the desired content.

```js
// Login
// ...

const { data: permissions } = await axios.get('/api/permissions');
const { data: roles } = await axios.get('/api/roles');

this.$laravel.setPermissions(permissions);
this.$laravel.setRoles(roles);
```

Do not forget to create the endpoints!

**Comment by muscle-rhnkyr** [comment]
@williamcruzme thanks for this! Now my question is when I set it up as you explain it and use with nuxt auth module

```
async login() {
        this.$nuxt.$loading.start()
        this.loading = false
        try {
          await this.$auth.loginWith('local', {
            data: {
              email: this.email,
              password: this.password,
            },
          }).then(async () => {
            const { data: permissions } = await this.$axios.get('/api/permissions')
            const { data: roles } = await this.$axios.get('/api/roles')

            await this.$laravel.setPermissions(permissions)
            await this.$laravel.setRoles(roles)
          })
        } catch (e) {
          this.$nuxt.$loading.finish()
          this.loading = true
          this.error = e.response
        }
      },
```
the think is when I use somewhere in another partial * template it seems it does not work. When I try `console.log(this.$laravel.getRoles()) `in that template's mounted method returns empty array. I am sure end point returns data.

* left-menu.vue 

`<template>

  
  <div id="layout-sidenav" class="layout-sidenav sidenav sidenav-vertical bg-dark">

    <div class="app-brand bg-white pt-2">
      <nuxt-link class="app-brand-text demo sidenav-text font-weight-normal ml-2" :to="'/'+role+'/'">
        <img
          class="auth-form__logo d-table mx-auto mb-3"
          src="xxxxxx"
          alt="MinjaCo" style="width: 9rem"
        >
      </nuxt-link>
      <a href="javascript:void(0)" class="layout-sidenav-toggle sidenav-link text-large ml-auto d-none">
        <i class="ion ion-md-menu align-middle"></i>
      </a>
    </div>

    <div class="sidenav-divider mt-0"></div>

    
    <ul class="sidenav-inner py-1 ps ps--active-y">

      
      <li class="sidenav-item" v-role="'customer-service'">
        <nuxt-link class="sidenav-link" exact-active-class="active" :to="'/'+role+'/'">
          <i class="sidenav-icon fa fa-home"></i>
          <div>{{$t('titles.overview')}}</div>
        </nuxt-link>
      </li>
      <li class="sidenav-item" v-role:any="'customer-service|admin'">
        <nuxt-link class="sidenav-link" exact-active-class="active" :to="'/'+role+'/customers'">
          <i class="sidenav-icon fa fa-users"></i>
          <div>{{$t('titles.customers')}}</div>
        </nuxt-link>
      </li>
      <li class="sidenav-item" v-role:any="'customer-service|admin'">
        <nuxt-link class="sidenav-link" exact-active-class="active" :to="'/'+role+'/orders'">
          <i class="sidenav-icon fa fa-shopping-bag"></i>
          <div>{{$t('titles.orders')}}</div>
        </nuxt-link>
      </li>
      
    </ul>
  </div>
  
</template>

< script >

  export default {

    mounted () {
      console.log(this.$laravel.getRoles())
    },

    computed: {

      role() {
        return this.$auth.user.role
      },
    },

  }
< /script >`

**Comment by williamcruzme** [comment]
@muscle-rhnkyr Permissions and roles are lost when you refresh the website. Is your website SPA or MPA?

**Comment by muscle-rhnkyr** [comment]
It is SPA.

**Comment by williamcruzme** [comment]
Thank you for your patience. I have published a new release (1.4.0) that has the option to persist the permissions and roles that you establish.

Greetings.

**Comment by muscle-rhnkyr** [comment]
Hey @williamcruzme . Thank you for that!!! Now works like a charm. Cheers.

**Comment by elsopi** [comment]
@williamcruzme thanks a lot for this library. Was facing the same issue, but it's fine now.

**Comment by williamcruzme** [comment]
@elsopi Thanks for your feedback!

**Comment by elsopi** [comment]
@williamcruzme for some reason, my app is no longer running properly and it started after adding persistence property.
This is the error message i get

`VM3497:1 Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.install (vendors.app.js:56617)
    at Function.Vue.use (commons.app.js:16578)
    at Module../plugins/libs.js (app.js:11830)
    at __webpack_require__ (runtime.js:791)
    at fn (runtime.js:151)
    at Module../.nuxt/index.js (app.js:3773)
    at __webpack_require__ (runtime.js:791)
    at fn (runtime.js:151)
    at Module.<anonymous> (app.js:1969)`

**Comment by williamcruzme** [comment]
@elsopi Make sure that at the moment you use the `setPermissions` and `setRoles` functions, the parameter value is not undefined.

**Comment by gladtoseeuhappy** [comment]
> Thank you for your patience. I have published a new release (1.4.0) that has the option to persist the permissions and roles that you establish.
> 
> Greetings.

 I already set option to persist but my side bar not work at all, except reload page

Vue.use(LaravelPermissions, { persistent: true });
